commandapi-platforms/commandapi-bukkit/commandapi-bukkit-core/src/main/java/dev/jorel/commandapi/CommandAPICommand.java

```diff
@@ -22,4 +23,27 @@ protected CommandAPICommand newConcreteCommandAPICommand(CommandMetaData<Command
 	public CommandAPICommand instance() {
 		return this;
 	}
+
+	/**
+	 * Registers the command with a given namespace
+	 *
+	 * @param namespace The namespace of this command. This cannot be null or empty
+	 *
+	 */
+	public void register(String namespace) {
+		if (CommandAPIBukkit.get().isInvalidNamespace(this.meta.commandName, namespace)) {
+			super.register();
+			return;
+		}
```

Comment:
We should make sure that no one sets the default namespace to an empty String on Bukkit. Otherwise, this plugin triggers a `StackOverflow` here.

```java
@Override
public void onLoad() {
    CommandAPI.onLoad(new CommandAPIBukkitConfig(this).setNamespace(""));
    new CommandAPICommand("test").register();
}
```

Exception: https://paste.gg/p/anonymous/b3070f86af3b4b57a5055eb86e289a8a

The same thing can happen using a `CommandTree`.

It would probably also be a good idea to add tests for this code path to make sure an invalid namespace really is replaced with the default namespace, and that the default namespace can be properly configured using `setNamespace` and `usePluginNamespace`.